src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -18,14 +18,19 @@ public static partial class JsonSerializer
             ref Utf8JsonReader reader,
             ref ReadStack state)
         {
-            if (state.Current.ObjectState < StackFrameObjectState.MetadataPropertyName)
+            if (state.Current.ObjectState < StackFrameObjectState.MetadataTryReadName)
             {
+                state.Current.ObjectState = StackFrameObjectState.MetadataReadName;
```

Comment:
Maybe we should rename `MetadataTryReadName` to `MetadataTryReadNameOrEndObject`.

Reply:
Can we add a helper method for all of these "try read and set next state"?